File: lib-core-fs/src/main/java/com/ustadmobile/core/impl/NoCacheResponse.java
package com.ustadmobile.core.impl;

import com.ustadmobile.core.impl.http.UmHttpResponse;

import java.io.InputStream;

public class NoCacheResponse extends AbstractCacheResponse {

    private UmHttpResponse networkResponse;

    public NoCacheResponse(UmHttpResponse networkResponse) {
        this.networkResponse = networkResponse;
        setCacheResponse(MISS);
    }

    @Override
    public String getFileUri() {
        return null;
    }

    @Override
    public boolean isFresh(int timeToLive) {
        return true;
    }

    @Override
    public boolean isFresh() {
        return true;
    }

    @Override
    public String getHeader(String headerName) {
        return networkResponse.getHeader(headerName);
    }

    @Override
    public byte[] getResponseBody() {
        return networkResponse.getResponseBody();
    }

    @Override
    public InputStream getResponseAsStream() {
        return networkResponse.getResponseAsStream();
    }

    @Override
    public boolean isSuccessful() {
        return networkResponse.isSuccessful();
    }

    @Override
    public int getStatus() {
        return networkResponse.getStatus();
    }
}

File: lib-core-fs/src/main/java/com/ustadmobile/core/impl/NoCacheResponse.kt
package com.ustadmobile.core.impl

import com.ustadmobile.core.impl.http.UmHttpResponse

import java.io.InputStream

class NoCacheResponse(private val networkResponse: UmHttpResponse) : AbstractCacheResponse() {

    override val fileUri: String?
        get() = null

    override val isFresh: Boolean
        get() = true

    override val responseBody: ByteArray?
        get() = networkResponse.responseBody

    override val responseAsStream: InputStream?
        get() = networkResponse.responseAsStream

    override val isSuccessful: Boolean
        get() = networkResponse.isSuccessful

    override val status: Int
        get() = networkResponse.status

    init {
        cacheResponse = AbstractCacheResponse.MISS
    }

    override fun isFresh(timeToLive: Int): Boolean {
        return true
    }

    override fun getHeader(headerName: String): String? {
        return networkResponse.getHeader(headerName)
    }
}
